### ISSUE TITLE
Clash with imenu configuration (Windows)

### ISSUE THREAD

**Issue by ivan-m** [magit/magit#3788]
I recently upgraded my version of magit from Melpa last week (after not having touched it since about November) and found that trying to view individual commits was no longer working: instead of showing a commit (by hitting `Enter` on a commit in `magit-status` or `magit-log`, thus running `magit-visit-ref`/`magit-show-commit`), Emacs would hang and I would need to kill it manually.  Note that nothing is seen at all in terms of the commit information, and `magit-process-buffer` is empty (even on successful commands).

(Note: this does not seem to be the case for every git repository: I have 2 repositories and it still worked in repo A whilst hanging in repo B; however, I also had a worktree checkout of repo A and it hangs there as well.  The only thing in common between repo B and the worktree checkout is the presence of a `.dir-locals.el` file that isn't setting anything magit- or imenu-specific.)

After bisecting my config, I tracked the change down to [this snippet](https://www.emacswiki.org/emacs/ImenuMode#toc2) that I've been using for years:

```elisp
 (defun try-to-add-imenu ()
  (condition-case nil (imenu-add-to-menubar "yourFancyName") (error nil)))
 (add-hook 'font-lock-mode-hook 'try-to-add-imenu)
```

If I run the output of `M-x magit-emacs-Q-command RET` then it works; if I evaluate the above snippet then - unlike with my actual config - the commit information starts to show (as in I have the author, parent, etc. but no actual diff output) before then Emacs hangs (just after a few prompts regarding the local variables list).  If I remove the `.dir-locals.el` file then I no longer get prompted, but then Emacs hangs immediately without even the initial meta-information about the commit appearing.

System information:

`magit-version`: Magit 20190311.1451, Git 2.17.0, Emacs 26.1, cygwin
OS: Windows 7; both Emacs and Git are installed via Cygwin.

**Comment by tarsius** [comment]
So if I understand correctly, removing *both* your snippet and `.dir-locals.el` fixes the issue? If so it would probably be a good idea to post the latter too.

**Comment by ivan-m** [comment]
No, just removing the snippet. Having the .dir-locals.el present just
allowed the buffer with metadata appear before it froze.


On mobile; please excuse any tpyos.

**Comment by tarsius** [comment]
Replace `condition-case` with `with-demoted-errors` to get useful information. Maybe even remove it altogether and enable `debug-on-error`.

**Comment by tarsius** [comment]
I get this error even if Magit is not loaded:

```
imenu-unavailable-error: imenu unavailable: "The mode `Help' does not support Imenu"
Error in post-command-hook (global-font-lock-mode-check-buffers): (imenu-unavailable "The mode `Help' does not support Imenu")
```
